### PR TITLE
Linkerd init container should be run first

### LINKED ISSUE #7999: Allow linkerd-init initContainer injection to be configurable via workload annotation
### What problem are you trying to solve?

Meshing databases while also meshing the initContainers that work against them (https://github.com/linkerd/linkerd2/issues/3812).

### How should the problem be solved?

Allow injection of the linkerd-init initContainer to optionally be done at the front of the array.

### Any alternatives you've considered?

We are going to shuffle our initContainers into the app itself, but considering this is a bog standard use case for initContainers, and the KEPs linked from #3812 were rejected, it seems reasonable that Linkerd support this use case.

Otherwise, we don't have much of a choice other than to weaken our mesh's security stance.

### How would users interact with this feature?

A `config.linkerd[...]` annotation seems reasonable.

### Would you like to work on this feature?

no

### PR DESCRIPTION
**Pull request by krzysztofdrys**
When there are multiple init containers, linkerd init container should
go first.

Fixes #7999

Signed-off-by: Krzysztof Dryś <krzysztofdrys@gmail.com>

### CODE REVIEW

**Comment by krzysztofdrys** [comment]
I am not sure, whether the change in integration tests adds much value. I added it initially, because I was unaware that this is tested in so many different places. 

Anyway, I am open for feedback.

File: test/integration/install/inject/inject_test.go [hunk 1]
```diff
@@ -481,29 +482,79 @@ func TestInjectAutoPod(t *testing.T) {
 			testutil.Fatalf(t, "expected pod in namespace %s to have %s opaque ports, but it had %s", ns, manualOpaquePorts, annotation)
 		}
 
+		initContainerPods, err := TestHelper.GetPods(ctx, ns, map[string]string{"app": initContainerPodName})
+		if err != nil {
+			testutil.AnnotatedFatalf(t, "failed to get pods", "failed to get pods for namespace %s: %s", ns, err)
+		}
+		if len(opaquePods) != 1 {
```
Comment (kleimkuhler) [inline test/integration/install/inject/inject_test.go:489]:
```suggestion
		if len(initContainerPods) != 1 {
```

Comment (krzysztofdrys) [reply test/integration/install/inject/inject_test.go]: Good catch!

Fixed in https://github.com/linkerd/linkerd2/pull/8034/commits/d34e55e5ea31f1d69f92017f2841fe6b27b63c48

**Comment by kleimkuhler** [comment]
@krzysztofdrys sorry for the delay in getting back to you on this. I just left a comment on #7999 to talk through some of the drawbacks this change may introduce. I'm curious in your thoughts on it; we should figure out what makes sense in the issue before coming back to this change.

**Comment by olix0r** [comment]
@krzysztofdrys We appreciate you taking this up, but I think we were wrong about this change. I'm going to close this out while we figure out a better long term solution to this problem.

**Pull request closed without merging**